django/db/models/expressions.py

```diff
@@ -547,6 +550,58 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None,
         return ResolvedOuterRef(self.name)
 
 
+class SliceableF(F):
+    """
+    An object that can add slicing notation to F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f_obj, slice_obj):
+        if not isinstance(slice_obj, (int, slice)):
```

Comment:
Using `int` is untested and doesn't work as expected, because it uses 1-based indexing instead of 0-based indexing.